woocommerce/woocommerce-android
# Coupons: Add Initial Coupon Restrictions Screen, Navigation, and Data Passing

Closes: #5532


### Description

This PR adds:
1. The initial/empty screen for Coupon Restrictions screen,
2. Navigation component setup between Edit Coupon Screen <--> Coupon Restrictions screen including data transfer,
3. Basic VM/Fragment/Compose component for Coupon Restrictions,
4. A new data class `CouponRestrictions` to better encapsulate the restrictions part of an `Coupon` instance.


### Testing instructions
1. Go to edit a Coupon,
2. Find the "Usage Restrictions" option and tap it,
3. Ensure the Coupon Restrictions screen is shown, with two sample inputs that don't really do anything:
4. Tap close or back button and ensure the Edit Coupon screen is shown again.


### Video

https://user-images.githubusercontent.com/266376/169013139-9d15976c-478e-45fa-88ba-04d576ab3c62.mov
